src/include/parser/postgresparser.h

```diff
@@ -11,9 +11,26 @@
 namespace terrier {
 namespace parser {
 
+class ParseResult {
+ public:
+  void AddStatement(std::unique_ptr<SQLStatement> statement) { statements_.emplace_back(std::move(statement)); }
+
+  void AddExpression(std::unique_ptr<AbstractExpression> expression) {
+    expressions_.emplace_back(std::move(expression));
+  }
+
+  const std::vector<std::unique_ptr<SQLStatement>> &GetStatements() { return statements_; }
+
+  const std::vector<std::unique_ptr<AbstractExpression>> &GetExpressions() { return expressions_; }
```

Comment:
Should this be returned as a std::vector<std::managed_ptr<AbstractExpression>>? From the looks of it when ParseResult get destructed expressions_ get's cleaned up so all the AbstractExpressions would also be cleaned up. So doesn't that mean ParseResult is managing this data?

Reply:
Yes